api/params.go

```diff
@@ -372,6 +374,11 @@ func (p *Params) SetDefaults(gitSource, gitOwner, gitName, appLabel, buildVersio
 		p.Container.Port = 5000
 	}
 
+	if p.Container.PortGrpc <= 0 {
+		// gRPC is optional, we have to opt-in by setting the portgrpc field of the deployment.
+		p.Container.PortGrpc = 0
```

Comment:
If the gRPC port is zero, then when we generate the Deployment YAML, we won't even include the extra container port, so that it's not created if not necessary.